dev/tasks/macros.jinja

```diff
@@ -221,3 +222,35 @@ on:
         cp ${formula} $(brew --repository homebrew/core)/Formula/
       done
 {% endmacro %}
+
+{%- macro change_r_pkg_version(date = '$(date +%Y%m%d)') -%}
+  - name: Modify version
+    shell: bash
+    run: |
+      cd arrow/r
+      sed -i.bak -E -e \
+        's/(^Version: [0-9]+\.[0-9]+\.[0-9]+).*$/\1.'"{{ date }}"'/' \
+        DESCRIPTION
+      head DESCRIPTION
+      rm -f DESCRIPTION.bak
```

Comment:
arrow-r-nightly has this, which is useful in order to be able to build binaries for the release branch. I'm not sure that this is the right way to trigger this, but adding this condition would allow us to support it. (If nothing else, we should leave a TODO here noting that we will need a way to build without appending a date to the version.)

```suggestion
      if [ "{{ date }}" != "" ]; then
        sed -i.bak -E -e \
          's/(^Version: [0-9]+\.[0-9]+\.[0-9]+).*$/\1.'"{{ date }}"'/' \
          DESCRIPTION
        head DESCRIPTION
        rm -f DESCRIPTION.bak
      fi
```

Reply:
I saw that logic in the setup step over in arrow-r-nightly, so you use that to build the cran release binaries? I will look into  how to integrate this in a way that makes it easy to trigger via crossbow.

Reply:
The possibility to provide a custom version is now also available in this workflow via a param so:
`@gha crossbow submit --param custom_version=8.5.3 r-nightly-packages`. (Only possible from a branch within apache/arrow, see my comment)